src/Microsoft.DotNet.Watcher.Core/Internal/FileWatcher/FileWatcherFactory.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.DotNet.Watcher.Core.Internal
+{
+    public static class FileWatcherFactory
+    {
+        public static IFileSystemWatcher CreateWatcher(string watchedDirectory)
+        {
+            var envVar = Environment.GetEnvironmentVariable("USE_POLLING_FILE_WATCHER");
```

Comment:
Are we using the same var for the cshtml PR that @pranavkm is doing? If so then `POLLING_FILE_WATCHER` probably isn't accurate. More like `USE_NETWORK_SHARE_FALLBACK` or `USE_NETWORK_SHARE_WATCHER` or something. If they are different keys then this is fine :)